src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/VariableDeclarationUsageDistanceCheck.java

```diff
@@ -835,15 +825,27 @@ private static DetailAST getFirstNodeInsideSwitchBlock(
     }
 
     /**
-     * Helper method for getFirstNodeInsideSwitchBlock to return the first CASE_GROUP or
-     * SWITCH_RULE ast.
+     * Helper method for getFirstNodeInsideSwitchBlock to return all variable
+     * usage expressions inside a given switch block.
      *
      * @param block the switch block to check.
-     * @return DetailAST of the first CASE_GROUP or SWITCH_RULE.
+     * @param variable variable which is checked for in switch block.
+     * @return List of usages or empty list if none are found.
      */
-    private static DetailAST getFirstCaseGroupOrSwitchRule(DetailAST block) {
-        return Optional.ofNullable(block.findFirstToken(TokenTypes.CASE_GROUP))
-            .orElseGet(() -> block.findFirstToken(TokenTypes.SWITCH_RULE));
+    private static List<DetailAST> getVariableUsageExpressionsInsideSwitchBlock(DetailAST block,
+                                                                            DetailAST variable) {
+        final List<DetailAST> variableUsageExpressions = new ArrayList<>();
+
+        for (int type : List.of(TokenTypes.CASE_GROUP, TokenTypes.SWITCH_RULE)) {
+            TokenUtil.forEachChild(block, type, child -> {
```

Comment:
Please share a problem to write util method that takes list of tokens

Reply:
TokenUtil.forEachChild(block, List.of(TokenTypes.CASE_GROUP, TokenTypes.SWITCH_RULE))

Reply:
Not a problem at all. I guess I did not pick up your thoughts/what you meant. On it.

Reply:
Done. Added method and a test for it.